django/template/context.py

```diff
@@ -175,6 +181,8 @@ def update(self, other_dict):
         "Pushes other_dict to the stack of dictionaries in the Context"
         if not hasattr(other_dict, '__getitem__'):
             raise TypeError('other_dict must be a mapping (dictionary-like) object.')
+        if isinstance(other_dict, BaseContext):
+            return ContextDict(self, other_dict.flatten())
```

Comment:
could we do `other_dict = other_dict.flatten()` so we only need the one return statement?